components/tidb_query/src/rpn_expr/impl_string.rs

```diff
@@ -49,6 +49,23 @@ pub fn ascii(arg: &Option<Bytes>) -> Result<Option<i64>> {
     }))
 }
 
+#[rpn_fn]
+#[inline]
+pub fn locate_binary_2_args(substr: &Option<Bytes>, s: &Option<Bytes>) -> Result<Option<i64>> {
+    let substr = match substr {
```

Comment:
It may be simpler to match the two arguments at the same time, like:

```rust
match (substr, s) {
    (Some(...), Some(...)) => { ... },
    _ => ...
}
```